src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,219 @@
-/**
- * Contains metadata relevant to a representation.
- */
-import type { Quad } from 'rdf-js';
+import { quad as createQuad, literal, namedNode } from '@rdfjs/data-model';
+import { Store } from 'n3';
+import type { BlankNode, Literal, NamedNode, Quad, Term } from 'rdf-js';
+import { CONTENT_TYPE } from '../../util/MetadataTypes';
+
+const shortcuts: { [id: string]: NamedNode } = {
+  contentType: CONTENT_TYPE,
+};
 
 /**
- * Metadata corresponding to a {@link Representation}.
+ * Stores the metadata triples and provides methods for easy access.
+ * Most functions return the metadata object to allow for chaining.
  */
-export interface RepresentationMetadata {
+export class RepresentationMetadata {
+  private store: Store;
+  private id: NamedNode | BlankNode;
+
+  /**
+   * @param identifier - Identifier of the resource relevant to this metadata.
+   *                     A blank node will be generated if none is provided.
+   *                     Strings will be converted to named nodes. @ignored
+   *
+   * `@ignored` tag is necessary for Components-Generator.js
+   */
+  public constructor(identifier?: NamedNode | BlankNode | string);
```

Comment:
I actually expected this one to have `metadata?: RepresentationMetadata` as a second optional argument. So you can specify both an identifier and metadata, omit either, or omit both.